### PR TITLE
Expose message:sent and message:received events

### PR DESCRIPTION
**Pull request by Mario54**
@alavers @lemieux @dannytranlx @mspensieri 

Client applications can now add event handlers to `message:sent` and `message:received` events and add any behavior they desire.

### CODE REVIEW

File: test/specs/chatController.spec.js [hunk 1]
```diff
@@ -97,6 +97,16 @@ describe('ChatController', function() {
                 done();
             }).catch(done);
         });
+
+        it('should trigger message:sent', function(done) {
```
Comment (mspensieri) [inline test/specs/chatController.spec.js]: `done` callback is icky.

```
it('should trigger message:sent', function() {
    messageSentSpy = sandbox.spy();
    chatController.once('message:sent', messageSentSpy);

    return chatController.sendMessage(message).then(function() {
        messageSentSpy.should.have.been.calledOnce;
    });
});
```

File: src/js/main.js [hunk 2]
```diff
@@ -298,6 +301,16 @@ _.extend(Smooch.prototype, Backbone.Events, {
         }.bind(this));
     },
 
+    _subscribeToEvents: function() {
+        this._chatController.on('message:received', _(function(message) {
```
Comment (lemieux) [inline src/js/main.js]: use `this.listenTo(this._chatController, 'message:received', ...)` instead. That way, if the chatController is destroy and reinitialize again, the old instance can be garbage collected and vice-versa, if the Smooch object is destroyed..

Comment (dannytranlx) [inline src/js/main.js]: thanks Marionette! :grin:

**Comment by lemieux** [comment]
:+1:

**Comment by mspensieri** [comment]
:+1:

**Pull request merged**